### PR TITLE
chore(ci): Generate docs dynamically in pre-commit hook

### PR DESCRIPTION
**Pull request by nalanj**
This will run `docs:generate` and then add updates to `docs-v2/snippets/generated` in our pre-commit hooks so that we'll keep docs up to date.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -2,3 +2,6 @@
 . "$(dirname -- "$0")/_/husky.sh"
 
 npx lint-staged --allow-empty
+
+npm run docs:generate
+git add ./docs-v2/snippets/generated/
```
Comment (bodinsamuel) [inline .husky/pre-commit]: can we just run it when stuff has changed inside docs-v2?

**Review by bodinsamuel — APPROVED** [review]

File: .husky/pre-commit (hunk 1 above)
Comment (nalanj) [reply .husky/pre-commit]: The task is what changes stuff inside of `docs-v2`.

Comment (bodinsamuel) [reply .husky/pre-commit]: not sure I get what you mean

Comment (nalanj) [reply .husky/pre-commit]: Nothing has changed inside `docs-v2` until we run the command

Comment (bodinsamuel) [reply .husky/pre-commit]: If I read the code correctly: it changes stuff if flows.yaml is modified (should be done in the other repo), a `provider:` has been changed, so it's a change in docs-v2. Am I missing something?

Comment (nalanj) [reply .husky/pre-commit]: Sure sure.

**Review by bodinsamuel — APPROVED** [review]

**Pull request merged**